wn/lmf.py

```diff
@@ -0,0 +1,639 @@
+        _assert_closed(event, elem, 'SenseRelation')
+        relations.append(_load_relation(elem, SenseRelation, _sense_rels))
+        event, elem = next(events)
+
+    examples: List[Example] = []
+    while event == 'start' and elem.tag == 'Example':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Example')
+        examples.append(_load_example(elem))
+        event, elem = next(events)
+
+    counts: List[Count] = []
+    while event == 'start' and elem.tag == 'Count':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Count')
+        counts.append(_load_count(elem))
+        event, elem = next(events)
+
+    _assert_closed(event, elem, 'Sense')
+
+    return Sense(
+        attrs['id'],
+        attrs['synset'],
+        relations=tuple(relations),
+        examples=tuple(examples),
+        counts=tuple(counts),
+        lexicalized=_get_bool(attrs.get('lexicalized', 'true')),
+        adjposition=_get_optional_literal(attrs.get('adjposition'), _adjpositions),
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+_R = TypeVar('_R', SynsetRelation, SenseRelation)
+
+
+def _load_relation(elem, cls: Type[_R], choices: Tuple[str, ...]) -> _R:
+    attrs = elem.attrib
+    return cls(
+        attrs['target'],
+        _get_literal(attrs['relType'], choices),
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+def _load_example(elem) -> Example:
+    attrs = elem.attrib
+    return Example(
+        elem.text,
+        language=attrs.get('language'),
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+def _load_count(elem) -> Count:
+    attrs = elem.attrib
+    value: int = -1
+    try:
+        value = int(elem.text)
+    except ValueError:
+        warnings.warn(f'count must be an integer: {elem.text}', LMFWarning)
+    return Count(
+        value,
+        status=attrs.get('status'),
+        note=attrs.get('note'),
+        confidence=_get_confidence(attrs),
+        dcmeta=_get_dublin_core_metadata(attrs),
+    )
+
+
+def _load_syntactic_behavior(elem) -> SyntacticBehaviour:
+    attrs = elem.attrib
+    return SyntacticBehaviour(
+        attrs['subcategorizationFrame'],
+        senses=tuple(attrs.get('senses', '').split()),
+    )
+
+
+def _load_synset(local_root, events) -> Synset:
+    attrs = local_root.attrib
+    event, elem = next(events)
+
+    definitions: List[Definition] = []
+    while event == 'start' and elem.tag == 'Definition':
+        event, elem = next(events)
+        _assert_closed(event, elem, 'Definition')
+        definitions.append(_load_definition(elem))
+        event, elem = next(events)
```

Comment:
These blocks look repetitive, maybe some wrapper around it?

```
        event, elem = next(events)
        _assert_closed(event, elem, 'xxx')
        xxx.append(_load_xxx(elem))
        event, elem = next(events)
```

Reply:
Possible, but would probably require another type variable so it type checks.